src/target/riscv/riscv.c

```
@@ -5588,6 +5588,52 @@ COMMAND_HANDLER(handle_riscv_virt2phys_mode)
 	return ERROR_OK;
 }
 
+COMMAND_HANDLER(riscv_smp_set_ext_triggers)
+{
+	struct target *target = get_current_target(CMD_CTX);
+	RISCV_INFO(r);
+
+	if (!target->smp) {
+		LOG_ERROR("Command cannot be used on a target that does not belong to any SMP group.");
+		return ERROR_COMMAND_SYNTAX_ERROR;
+	}
```

1) This is not a syntax error. Please use `return ERROR_FAIL` (or similar) instead of `return ERROR_COMMAND_SYNTAX_ERROR`.

2) Please move this check down - first check all syntax errors, then other errors.

3) Please use `LOG_TARGET_ERROR`. The message could be: `This target does not belong to any SMP group. Command 'riscv smp_set_ext_triggers' cannot be used`

---

Addressed.